sdks/python/src/opik/file_upload/upload_client.py

```diff
@@ -0,0 +1,105 @@
+        self, rest_client: rest_api_client.OpikApi, httpx_client: httpx.Client
+    ) -> None:
+        self.rest_client = rest_client
+        self.httpx_client = httpx_client
+
+    def start_upload(
+        self,
+        upload_options: file_upload_options.FileUploadOptions,
+        num_of_file_parts: int,
+        base_url_path: str,
+    ) -> MultipartUploadMetadata:
+        """Starts upload by sending request to the backend and receiving upload metadata. The upload metadata
+        will include the list of pre-signed URLs for direct S3 upload and upload ID assigned to this file
+        upload operation. If backend decides to force upload to then local endpoint then list of URLs will
+        include only one URL and upload ID will have magic value 'BEMinIO'."""
+        response = self.rest_client.attachments.start_multi_part_upload(
+            file_name=upload_options.file_name,
+            num_of_file_parts=num_of_file_parts,
+            entity_type=upload_options.entity_type,
+            entity_id=upload_options.entity_id,
+            path=base_url_path,
+            mime_type=upload_options.mime_type,
+            project_name=upload_options.project_name,
+        )
+        return MultipartUploadMetadata(
+            upload_id=response.upload_id, urls=response.pre_sign_urls
+        )
+
+    def s3_upload_completed(
+        self,
+        file_size: int,
+        upload_options: file_upload_options.FileUploadOptions,
+        upload_metadata: MultipartUploadMetadata,
+        file_parts: List[rest_api_types.MultipartUploadPart],
+    ) -> None:
+        """Invoked to finalize direct S3 file upload operation on the backend. It is invoked after all file parts
+        was successfully uploaded to S3."""
+        self.rest_client.attachments.complete_multi_part_upload(
+            file_name=upload_options.file_name,
+            entity_type=upload_options.entity_type,
+            entity_id=upload_options.entity_id,
+            file_size=file_size,
+            upload_id=upload_metadata.upload_id,
+            uploaded_file_parts=file_parts,
+            project_name=upload_options.project_name,
+            mime_type=upload_options.mime_type,
+        )
+
+    @retry_decorator.opik_rest_retry
+    def upload_file_local(
+        self,
+        upload_url: str,
+        file_path: str,
+        chunk_size: int = -1,
+        monitor: Optional[upload_monitor.FileUploadMonitor] = None,
+    ) -> None:
+        """Invoked to upload file to the local backend using httpx client configured with necessary authorization
+        headers. Raises the `HTTPStatusError` if one occurred."""
+        response = self.httpx_client.put(
+            url=upload_url,
+            content=_data_generator(file_path, chunk_size=chunk_size, monitor=monitor),
+        )
+        response.raise_for_status()
+
+
+def _data_generator(
+    file_path: str,
+    chunk_size: int = -1,
+    monitor: Optional[upload_monitor.FileUploadMonitor] = None,
```

Comment:
Let's remove the default value for the monitor. There is already one on the upper level; and there is no need to have multiple since it can lead to unexpected bugs.